AssertionError during lexical analysis of a macro definition

## Compiler version

3.8.3-RC3

## Minimized code

When facing this code, the compiler should emit a syntax error because a macro parameter cannot be a quasiquote patterns:

```scala
object Crash {
  def macroImpl(using Quotes)(q"a: String"): Expr[Boolean] = 1    
}
```

Instead, an assertion error occurs:

```scala
Exception in thread "main" java.lang.AssertionError: assertion failed
	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:13)
	at dotty.tools.dotc.parsing.Scanners$Scanner.lookahead(Scanners.scala:1126)
	at dotty.tools.dotc.parsing.Parsers$Parser.termParamClause$$anonfun$1(Parsers.scala:3824)
	...
	at dotty.tools.repl.Main.main(Main.scala)
```

_Thanks to OSTIF/Quarkslab for finding this._

--- a/compiler/src/dotty/tools/dotc/parsing/Parsers.scala
+++ b/compiler/src/dotty/tools/dotc/parsing/Parsers.scala
@@ -3859,7 +3859,7 @@ object Parsers {
                 syntaxError(em"`using` expected")
               val (firstParamMod, paramsAreNamed) =
                 var mods = EmptyModifiers
-                if in.lookahead.isColon then
+                if in.token != INTERPOLATIONID && in.lookahead.isColon then
                   (mods, true)
                 else if isConsume then (mods, true)
                 else

--- a/tests/neg/i25717.scala
+++ b/tests/neg/i25717.scala
@@ -1,0 +1,9 @@
+import scala.quoted.*
+
+object Crash {
+  def macroImpl(using Quotes)(q"a: String"): Expr[Boolean] = ??? // error
+
+  def f(s"a: String") = ??? // error
+
+  def softie(into q"a: String") = ??? // error
+}

